athena-cloudwatch/src/main/java/com/amazonaws/athena/connectors/cloudwatch/CloudwatchMetadataHandler.java

```diff
@@ -325,6 +351,40 @@ public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest
         return new GetSplitsResponse(request.getCatalogName(), splits, null);
     }
 
+    @Override
+    public GetDataSourceCapabilitiesResponse doGetDataSourceCapabilities(BlockAllocator allocator, GetDataSourceCapabilitiesRequest request)
+    {
+        ImmutableMap.Builder<String, List<OptimizationSubType>> capabilities = ImmutableMap.builder();
+        queryPassthrough.addQueryPassthroughCapabilityIfEnabled(capabilities, configOptions);
+
+        return new GetDataSourceCapabilitiesResponse(request.getCatalogName(), capabilities.build());
+    }
+
+    @Override
+    public GetTableResponse doGetQueryPassthroughSchema(BlockAllocator allocator, GetTableRequest request) throws Exception
+    {
+        if (!request.isQueryPassthrough()) {
+            throw new IllegalArgumentException("No Query passed through [{}]" + request);
+        }
+        // to get column names with limit 1
+        GetQueryResultsResult getQueryResultsResult = getResult(invoker, awsLogs, request.getQueryPassthroughArguments(), 1);
+        SchemaBuilder schemaBuilder = SchemaBuilder.newBuilder();
+        if (!getQueryResultsResult.getResults().isEmpty()) {
+            for (ResultField field : getQueryResultsResult.getResults().get(0)) {
+                schemaBuilder.addField(field.getField(), Types.MinorType.VARCHAR.getType());
+            }
+        }
+        else {
+            return new GetTableResponse(request.getCatalogName(),
+                    request.getTableName(),
+                    CLOUDWATCH_SCHEMA);
```

Comment:
Not sure if i like the idea of return basic schema for GetQueryPassThrough if we failed to get qpt result. Any particular reason on why we are doing this?

Reply:
not any particular reason. it is used just to provide similar behaviour as that of getTable method. I have removed that.